### PR TITLE
[wip] Add search schema

### PR DESCRIPTION
**Pull request by dblandin**
Add search schema to Metaphysics.

TODO: Implement search data loader

2019-01-29-124850_634x516_scrot
2019-01-29-124830_624x561_scrot

### CODE REVIEW

File: src/schema/search.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+      type: new GraphQLList(SearchResultItemType),
+      description: "Object associated with search result",
+    },
+  },
+})
+
+export const Search = {
+  type: new GraphQLList(SearchResult),
+  description: "Global search",
+  args: SearchArgs,
```
Comment (damassi) [inline src/schema/search.ts]: Maybe `searchArgs`? CapsCase is idiomatically reserved for types and classes.

**Comment by mzikherman** [comment]
🔥

File: src/schema/search.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    },
+    item: {
+      type: new GraphQLList(SearchResultItemType),
+      description: "Object associated with search result",
+    },
+  },
+})
+
+export const Search = {
+  type: new GraphQLList(SearchResult),
```
Comment (mzikherman) [inline src/schema/search.ts]: note this should be a connection, there should be a `searchResultConnection` type in here somewhere, like:

```
const searchResultConnection = connectionWithCursorInfo(SearchResult.type)
```

**Comment by mzikherman** [comment]
Overall, I like this. It offers the custom JSON immediately available as a result, and fuller fetching/resolving of the item can be found in `item`, which is a union of supported types. Seems reasonable/simple!

**Comment by alloy** [comment]
I figured it would be helpful to show examples of each PR.

```graphql
{
  search(query: "Banksy") {
    edges {
      node {
        # if you only want generic data
        label
        # if you want specific data for each concrete type
        item {
          ... on Artist {
            name
          }
          ... on Artwork {
            title
          }
        }
      }
    }
  }
}
```

**Comment by dblandin** [comment]
Thanks for the discussion! I'm going to go ahead and close this PR in favor of https://github.com/artsy/metaphysics/pull/1476.

Sounds like that direction hits a great balance between interface design and performance. :+1:

**Pull request closed without merging**